aries/aries_vcx/src/protocols/did_exchange/state_machine/responder/response_sent/mod.rs

```diff
@@ -28,35 +32,57 @@ impl DidExchangeResponder<ResponseSent> {
         request: Request,
         our_peer_did: &PeerDid<Numalgo4>,
         invitation_key: Option<Key>,
-    ) -> Result<TransitionResult<DidExchangeResponder<ResponseSent>, Response>, AriesVcxError> {
-        info!(
-            "DidExchangeResponder<ResponseSent>::receive_request >> request: {}, our_peer_did: \
+    ) -> Result<TransitionResult<DidExchangeResponder<ResponseSent>, AnyResponse>, AriesVcxError>
+    {
+        debug!(
+            "DidExchangeResponder<ResponseSent>::receive_request >> request: {:?}, our_peer_did: \
              {}, invitation_key: {:?}",
             request, our_peer_did, invitation_key
         );
+        let version = request.get_version();
+
         let their_ddo = resolve_ddo_from_request(&resolver_registry, &request).await?;
         let our_did_document = our_peer_did.resolve_did_doc()?;
-        // TODO: Check amendment made to did-exchange protocol in terms of rotating keys.
-        //       When keys are rotated, there's a new decorator which conveys that
-        let ddo_attachment_unsigned = ddo_to_attach(our_did_document.clone())?;
-        let ddo_attachment = match invitation_key {
+
+        let unsigned_attachment = match version {
+            DidExchangeTypeV1::V1_1(_) => assemble_did_rotate_attachment(our_peer_did.did()),
+            DidExchangeTypeV1::V1_0(_) => ddo_to_attach(our_did_document.clone())?,
+        };
+        let attachment = match invitation_key {
+            Some(invitation_key) => {
+                // TODO: this must happen only if we rotate DID; We currently do that always
+                //       can skip signing if we don't rotate did document (unique p2p invitations
+                //       with peer DIDs)
+                jws_sign_attach(unsigned_attachment, invitation_key, wallet).await?
+            }
             None => {
                 // TODO: not signing if invitation_key is not provided, that would be case for
                 //       implicit invitations. However we should probably sign with
                 //       the key the request used as recipient_vk to anoncrypt the request
                 //       So argument "invitation_key" should be required
-                ddo_attachment_unsigned
-            }
-            Some(invitation_key) => {
-                // TODO: this must happen only if we rotate DID; We currently do that always
-                //       can skip signing if we don't rotate did document (unique p2p invitations
-                //       with peer DIDs)
-                jws_sign_attach(ddo_attachment_unsigned, invitation_key, wallet).await?
+                unsigned_attachment
```

Comment:
Given the following from the DID Exchange 1.1 RFC:
> The `did_rotate~attach` attribute is optional, but SHOULD be included if the did attribute is resolvable and the `did_doc~attach` is not included. The value is the Base64url encoded DID, and signed with the key used in the invitation.

How is an implicit invitation (just having a resolvable DID that a `requester` uses to sends a DIDExchange Request to the `responder`) compatible with the DID Exchange 1.1? Do we essentially just ignore the SHOULD value here since there is no invitation? :thinking:

Reply:
IMO, it'd be good if we always require an `invitation_key` to be passed in. Even in the case of the implicit invitation, i believe that ideally the responder will sign the did_rotate with the service/recipient key associated with their public DID that received the request. 

^ This is what ACApy does when it receives a request as a result of an implicit invitation; the response did_doc/did_rotate is still signed

Reply:
> IMO, it'd be good if we always require an invitation_key to be passed in

i didn't change that here, as it was already like that, but maybe i should just change `invitation_key` to be required (not Option)

Reply:
Yeah that'd make sense to me
> i didn't change that here, as it was already like that, but maybe i should just change invitation_key to be required (not Option)

Didn't mean to bring up unrelated stuff necessarily, but I noticed it next to the changes in this PR. I think it'd be appropriate to require it.

Reply:
ok nvm i've moved this to a new issue. it requires a bit more work than this PR should handle IMO: https://github.com/hyperledger/aries-vcx/issues/1245

Reply:
Makes sense to me :+1: